src/libraries/System.Private.CoreLib/src/System/Threading/ManualResetEventSlim.cs

```diff
@@ -558,6 +559,9 @@ public bool Wait(int millisecondsTimeout, CancellationToken cancellationToken)
                             // If we see IsSet=false, then we are guaranteed that Set() will see that we are
                             // waiting and will pulse the monitor correctly.
 
+                            if (OperatingSystem.IsBrowser())
+                                return false;
+
```

Comment:
Added this because of `Monitor.Wait(m_lock, realMillisecondsTimeout)` on row 577, looks like it's safe to return false in case could now wait.

Reply:
This is not safe.  The `void Wait()` overload still ends up here, and if this returns false, the caller (which ignores the return value because it should never return false) will end up seeing `Wait()` successfully return even though the event may not have been signaled.